bin/templates/scripts/cordova/lib/check_reqs.js

```diff
@@ -121,6 +121,20 @@ module.exports.check_cocoapods = function (toolChecker) {
         // check whether the cocoapods repo has been synced through `pod repo` command
         // a value of '0 repos' means it hasn't been synced
         .then(function (toolOptions) {
+            let podVersion = shell.exec('pod --version', { silent: true });
+            if (podVersion) {
+                const semver = podVersion.toString().split('.');
+                if (semver.length >= 3) {
+                    const major = parseInt(semver[0]);
+                    const minor = parseInt(semver[1]);
+                    const patch = parseInt(semver[2]);
+
+                    // starting with 1.8.0 cocoapods now use cdn and we dont need to sync first
+                    if ((major >= 1 && minor >= 8 && patch >= 0) || toolOptions.ignore) {
+                        return Q.resolve(toolOptions);
+                    }
```

Comment:
I would recommend two thing.

1. Use the `semver` dependency that we already include with `cordova-ios`.
2. Return a native promise instead of using `Q`. We are removing the `Q` dependency in next major.

Example:

```js
const semver = require('semver');

if (semver.satisfies(`>=${podVersion.toString()}`) || toolOptions.ignore) {
    return Promise.resolve(toolOptions)
}

```

Reply:
Regarding 2. (Q vs native Promise): Here we return inside of a `then` callback. Anything we return will be wrapped inside a Promise automatically. So no need to manually wrap it at all here.

Reply:
Thanks, guys. I missed that this dependency was already on the tree. I've already changed the code and squashed the commit. I changed the other flow inside this method to return native promise also, to standardize this at least inside this method. Thanks!